### ISSUE TITLE
ModuleNotFoundError: No module named 'hdbscan'

### ISSUE THREAD

**Issue by chyumin** [SpikeInterface/spikeinterface#771]
`hdbscan` seems to be required now, but it's not listed in `requirements_full.txt`

I'm openning issue instead of a PR because I'm not sure if this is intended to be required or we need to change the code to make it optional again

`ModuleNotFoundError` raises when importing spikeinterface.full:

```
import spikeinterface.full as si
```

Full Traceback:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/yu/CatalystNeuro/spikeinterface/spikeinterface/full.py", line 15, in <module>
    from .sorters import *
  File "/home/yu/CatalystNeuro/spikeinterface/spikeinterface/sorters/__init__.py", line 2, in <module>
    from .sorterlist import *
  File "/home/yu/CatalystNeuro/spikeinterface/spikeinterface/sorters/sorterlist.py", line 19, in <module>
    from .spyking_circus2 import Spykingcircus2Sorter
  File "/home/yu/CatalystNeuro/spikeinterface/spikeinterface/sorters/spyking_circus2/__init__.py", line 1, in <module>
    from .spyking_circus2 import Spykingcircus2Sorter
  File "/home/yu/CatalystNeuro/spikeinterface/spikeinterface/sorters/spyking_circus2/spyking_circus2.py", line 5, in <module>
    from spikeinterface.sortingcomponents.clustering import find_cluster_from_peaks
  File "/home/yu/CatalystNeuro/spikeinterface/spikeinterface/sortingcomponents/clustering/__init__.py", line 1, in <module>
    from .method_list import clustering_methods
  File "/home/yu/CatalystNeuro/spikeinterface/spikeinterface/sortingcomponents/clustering/method_list.py", line 4, in <module>
    from .sliding_nn import SlidingNNClustering
  File "/home/yu/CatalystNeuro/spikeinterface/spikeinterface/sortingcomponents/clustering/sliding_nn.py", line 19, in <module>
    import hdbscan
ModuleNotFoundError: No module named 'hdbscan'

```

**Comment by samuelgarcia** [comment]
I see, the new spyking_circus need hdbscan.
@yger : can you make a patch in `is_installed()` that depend on hdbscan installed or not.

And also you should import all component in the function not at module level.

**Comment by yger** [comment]
what is the is_installed() method you are talking about?

**Comment by chyumin** [comment]
@yger awesome. Thank you

**Comment by chyumin** [comment]
Solved in #772
